### PR TITLE
Allow implementations of AbstractRowEncoder to support Structural type

### PR DESCRIPTION
**Pull request by Praveen2112**

### CODE REVIEW

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/AbstractRowEncoder.java [hunk 1]
```diff
@@ -118,8 +122,17 @@ else if (type instanceof TimestampType) {
...
+        }
+        else if (type instanceof MapType) {
+            appendMap((Map<Object, Object>) type.getObjectValue(session, block, position));
+        }
+        else if (type instanceof RowType) {
+            appendRow((List<Object>) type.getObjectValue(session, block, position));
+        }
         else {
-            throw new UnsupportedOperationException(format("Column '%s' does not support 'null' value", columnHandles.get(currentColumnIndex).getName()));
+            throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", type, columnHandles.get(currentColumnIndex).getName()));
```
Comment (kokosing) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/AbstractRowEncoder.java:135]: This error message could be extracted to separate commit.

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/AbstractRowEncoder.java [hunk 2]
```diff
@@ -18,6 +18,9 @@
 import com.google.common.primitives.SignedBytes;
```
Comment (kokosing) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/AbstractRowEncoder.java:18]: Commit message is a bit misleading. You didn't add support for structural types, but you allowed to implementations of `AbstractRowEncoder` to support them.

[optional] It would be nice use it, how about insert of structural types in Avro for Kafka?

**Review by hashhar — COMMENTED** [review]
The commit message is misleading since this only allows the implementations to add support for such types.

Looks good otherwise. A followup to implement these methods in Avro/JSON encoders will be nice.

**Review by ssheikin — APPROVED** [review]
Ok, after @kokosing @hashhar comments are implemented.

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/AbstractRowEncoder.java (hunk 2 above)
Comment (Praveen2112) [reply presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/AbstractRowEncoder.java:18]: > [optional] It would be nice use it, how about insert of structural types in Avro for Kafka?

Will be raising a follow up PR for implementation in Avro and JSON

**Review by hashhar — APPROVED** [review]

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/AbstractRowEncoder.java [hunk 3]
```diff
@@ -201,6 +214,21 @@ protected void appendSqlTimestampWithTimeZone(SqlTimestampWithTimeZone value)
...
+    }
+
+    protected void appendMap(Map<Object, Object> value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", value.getClass().getName(), columnHandles.get(currentColumnIndex).getName()));
+    }
+
+    protected void appendRow(List<Object> value)
+    {
+        throw new UnsupportedOperationException(format("Unsupported type '%s' for column '%s'", value.getClass().getName(), columnHandles.get(currentColumnIndex).getName()));
```
Comment (kokosing) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/AbstractRowEncoder.java:229]: you could extract a method in separate PR, as this would change the entire convention in this clas

**Review by kokosing — APPROVED** [review]

**Comment by kokosing** [comment]
Merged, thanks!
